ext/java/org/msgpack/jruby/Encoder.java

```diff
@@ -338,6 +338,29 @@ private void appendExtensionValue(ExtensionValue object) {
     buffer.put(payloadBytes.unsafeBytes(), payloadBytes.begin(), payloadSize);
   }
 
+  private void appendExtensionValue(ExtensionValue object) {
+    long type = ((RubyFixnum)object.get_type()).getLongValue();
+    if (type < -128 || type > 127) {
+	    throw object.getRuntime().newRangeError(String.format("integer %d too big to convert to `signed char'", type));
+    }
+    ByteList payloadBytes = ((RubyString)object.payload()).getByteList();
+    appendExt((int) type, payloadBytes);
+  }
+
+  private void appendOther(IRubyObject object, IRubyObject destination) {
+    if (this.registry != null) {
+      IRubyObject[] pair = this.registry.lookup(object.getType());
+      if (pair != null) {
+        RubyString bytes = pair[0].callMethod(runtime.getCurrentContext(), "call", object).asString();
+        int type = (int) ((RubyFixnum) pair[1]).getLongValue();
+        appendExt(type, bytes.getByteList());
+        return;
+      }
+    }
+    // registry is null or type is not registered
```

Comment:
I don't like comments in code, they are usually incorrect and no one ever removes them when the code changes. It's better to just not have them in the first place. If you feel that the code doesn't explain what the comment says then wrap the code in a method with a name that explains what's going on.